wasm_proto/core/util.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * `util.c` defines some utility functions
+ */
+
+#include "core.h"
+#include "util.h"
+#include "uint.h"
+
+void setcc(uint rslt) {
+	uint cc = 0x2;
+	if(rslt > 0) {
+		cc >>= 1;
+	} else if(rslt < 0) {
+		cc <<= 1;
+	}
+
+	core.psr = core.psr & ~0x7 | cc;
+}
+
+uint sext16(uint val, uint w) {
```

Comment:
does this sign extend n bits (with leading zeros) to 16 bits? it's a little unclear what this function does at a glance (since the logic is dense) - can you add a comment or make a more descriptive function name?

Reply:
I will add a comment.